client/src/modules/lead/EditLead/EditLeadContent/EditLeadTabs/EditLeadTabs.tsx

```diff
@@ -69,27 +73,59 @@ class EditLeadTabs extends React.Component<Props, State> {
   }
 
   public saveNote = (noteText: string) => {
-    const { editLead } = this.props;
-    const note = {
-      contact: editLead.contact ? editLead.contact._id : undefined,
-      lead: editLead._id,
-      organization: editLead.organization ? editLead.organization._id : undefined,
-      text: noteText,
-      user: this.props.userId,
-    };
+    let note: Note;
+    switch (this.props.modelType) {
+      case LEAD:
+        note = {
+          lead: this.props.model._id,
+          organization: this.props.model.organization ? this.props.model.organization._id : undefined,
+          contact: this.props.model.contact ? this.props.model.contact._id : undefined,
+          text: noteText,
+          user: this.props.userId,
+        };
+        break;
+      case CONTACT:
+        note = {
+          organization: this.props.model.organization ? this.props.model.organization._id : undefined,
+          contact: this.props.model._id,
+          text: noteText,
+          user: this.props.userId,
+        };
+        break;
+      case ORGANIZATION:
+      note = {
+        organization: this.props.model._id,
+        text: noteText,
+        user: this.props.userId,
+      };
+        break;
+      default:
+        note = {
+          text: noteText,
+          user: this.props.userId,
+        };
+        break;
+    }
     this.props.createNote(note);
     this.toggleFakeInput();
   }
 
   public saveActivity = (activity: Activity) => {
```

Comment:
move this.props.userId and this.props.model to const and remove redundant this.props parts